integration-tests/src/bin/gen_blockchain_data.rs

```diff
@@ -1,40 +1,69 @@
-            TransactionRequest, U256, U64,
+            transaction::eip2718::TypedTransaction, Address, GethDebugTracingOptions,
+            TransactionReceipt, TransactionRequest, H160, U256, U64,
         },
         utils::WEI_IN_ETHER,
     },
-    middleware::SignerMiddleware,
-    providers::{Middleware, PendingTransaction},
+    middleware::{
+        contract::{ContractDeploymentTx, ContractInstance},
+        SignerMiddleware,
+    },
+    providers::{Http, Middleware, PendingTransaction, Provider},
     signers::Signer,
-    solc::{CompilerInput, CompilerOutput, EvmVersion, Solc},
 };
 use integration_tests::{
-    get_client, get_provider, get_wallet, log_init, CompiledContract, GenDataOutput, CONTRACTS,
-    CONTRACTS_PATH, WARN,
+    bindings_benchmarks::{benchmarks::benchmarks, Len},
+    bindings_greeter::greeter::greeter,
+    bindings_openzeppelinerc20testtoken::openzeppelinerc_2_0testtoken::openzeppelinerc20testtoken as ozerc20tt,
+    get_client, get_provider, get_wallet, log_init, GenDataOutput,
 };
+
 use log::{error, info};
-use serde::de::Deserialize;
-use std::{collections::HashMap, fs::File, path::Path, sync::Arc, thread::sleep, time::Duration};
+use std::{collections::HashMap, fs::File, sync::Arc, thread::sleep, time::Duration};
 
-async fn deploy<T, M>(prov: Arc<M>, compiled: &CompiledContract, args: T) -> Contract<M>
+async fn deploy<'a, M, C, B>(
+    deployer_tx: ContractDeploymentTx<B, M, C>,
+    name: &'a str,
+) -> (Abi, H160, U64, C)
 where
-    T: Tokenize,
-    M: Middleware,
+    M: Middleware + 'a,
+    B: Borrow<M> + Clone + 'a,
+    C: From<ContractInstance<B, M>> + 'a,
 {
-    info!("Deploying {}...", compiled.name);
-    let factory = ContractFactory::new(compiled.abi.clone(), compiled.bin.clone(), prov);
-    factory
-        .deploy(args)
-        .expect("cannot deploy")
-        .confirmations(0usize)
-        .send()
+    info!("Deploying {}...", name);
+    let binding = deployer_tx.clone();
+    let contract_abi = binding.abi();
+    let c_abi = contract_abi.clone();
+    let contract_deployed = deployer_tx
+        .send_with_receipt()
         .await
-        .expect("cannot confirm deploy")
+        .expect("Could not confirm contract deployment");
+    let contract_instance = contract_deployed.0;
+    let deploy_tx_receipt = contract_deployed.1;
+    let block_number = deploy_tx_receipt
+        .block_number
+        .expect("failed to get block number");
+    let contract_address = deploy_tx_receipt
+        .contract_address
+        .expect("failed to get contract address");
+    (c_abi, contract_address, block_number, contract_instance)
+}
+
+#[allow(dead_code)]
+async fn dump_tx_trace(prov: Provider<Http>, receipt: TransactionReceipt, name: &str) {
+    let options = GethDebugTracingOptions::default();
+    let trace = prov
+        .debug_trace_transaction(receipt.transaction_hash, options)
+        .await
+        .expect("SOME REASON");
```

Comment:
Maybe this was meant to be replaced with some reason? :laughing:

Reply:
thanks :). >>> .expect("Failed to get transaction debug trace");